pkg/landscaper/gardenlet/apis/imports/v1alpha1/types.go

```diff
@@ -0,0 +1,94 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	gardenletconfigv1alpha1 "github.com/gardener/gardener/pkg/gardenlet/apis/config/v1alpha1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// LandscaperGardenletImport defines the landscaper import for the Gardenlet.
+// structure defined in Blueprint
+type LandscaperGardenletImport struct {
+	metav1.TypeMeta `json:",inline"`
+	// RuntimeCluster is the landscaper target containing the kubeconfig for the cluster
+	// where the Gardenlet should be deployed.
+	// This is the Kubernetes cluster targeted as Seed (via in-cluster mounted service account token),
+	// if not otherwise specified in `.componentConfiguration.seedClientConnection.kubeconfig`.
+	RuntimeCluster Target `json:"runtimeCluster"`
+	// GardenCluster is the landscaper target containing the kubeconfig for the
+	// Garden cluster (having Gardener resource groups!)
+	GardenCluster Target `json:"gardenCluster"`
+	// ImageVectorOverwrite contains the image vector override
+	ImageVectorOverwrite *runtime.RawExtension `json:"imageVectorOverwrite,omitempty"`
+	// ImageVectorOverwrite contains the image vector override for components deployed by the gardenlet
+	ComponentImageVectorOverwrites *runtime.RawExtension `json:"componentImageVectorOverwrites,omitempty"`
+	// SeedBackup contains configuration for an optional backup provider for the Seed cluster registered by the Gardenlet
+	// required when gardenlet.componentConfiguration.seedConfig.backup.secretRef is not set
+	// backup secret is deployed into the Garden cluster
+	SeedBackup *SeedBackup `json:"seedBackup,omitempty"`
+	// RevisionHistoryLimit is the revision history limit for the Gardenlet deployment
+	// Defaults to 10
+	RevisionHistoryLimit *int32 `json:"revisionHistoryLimit,omitempty"`
+	// Resources are the resource requirements for the Gardenlet pod
+	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
+	// ComponentConfiguration is the Gardenlet component configuration
+	ComponentConfiguration gardenletconfigv1alpha1.GardenletConfiguration `json:"componentConfiguration"`
+}
+
+// SeedBackup contains configuration for an optional backup provider for the Seed cluster registered by the Gardenlet
+type SeedBackup struct {
+	// Provider is the provider name {aws,gcp,...}
+	Provider string `json:"provider"`
+	// Credentials contains provider specific credentials
+	// Please check the documentation of the respective extension provider for the concrete format
+	Credentials *runtime.RawExtension `json:"credentials"`
```

Comment:
Shouldn't runtime.RawExtension be used for versioned objects like
```
apiVersion: mytype.com/v1
type: CustomType
mydata: abc
```
Otherwise you could also use json.RawMessage